src/Model/Revision.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Model;
+
+final class Revision
+{
+    /**
+     * @var int|string
+     */
+    private $id;
+
+    /**
+     * @var \DateTime
+     */
+    private $timestamp;
+
+    /**
+     * @var string
+     */
+    private $username;
+
+    /**
+     * @param int|string $id
+     */
+    public function __construct($id, \DateTime $timestamp, string $username)
```

Comment:
IMO, we should use `\DateTimeImmutable` if possible, since this timestamp is not intended to be changed.
```suggestion
    public function __construct($id, \DateTimeImmutable $timestamp, string $username)
```

Reply:
Currently, `$revision->getTimestamp()` is returning a DateTime, I copied this behavior.

Wouldn't it better to use the interface, `DateTimeInterface` instead ?

Reply:
If we aren't able to narrow the type by now, I think it is the best option we have.

Reply:
Done, I also provided better named getter.
After the 4.0 release, I will deprecate the old ones.
But here it's better to keep old ones for people who overrides only the templates.